Feat: Added support for Infinity Embeddings

Pratik960 opened the pull request:
This PR introduces:

- Introduces InfinityEmbeddings class in chonkie.embeddings

- Supports configurable model selection, automatic batching, and both sync (embed/embed_batch) and async (aembed/aembed_batch) methods

- Handles HTTP requests via requests and aiohttp, with clear error messages on non-200 responses

- Implements _prepare_batches to sort and chunk inputs and restore original order

Pratik960 commented:
@chonknick  PTAL :)

chonknick commented:
Hey @Pratik960! 😄

Thanks for the PR~ I'll review the code by this week. Meanwhile, if you could add some tests for the `InfinityEmbeddings` that would be pretty cool. You can make a file in the `tests/embeddings` directory with the name `test_infinity_embeddings.py` for it. 

Thanks 😊

Pratik960 commented:
Sure consider it done :)

Pratik960 commented:
@chonknick  Done the required changes!!

Pratik960 commented:
Hi @chonknick ,
i was getting the error that's why i have added those changes in unassociated files.
Anyways i have removed those changes and will create a new PR if needed.

Pratik960 commented:
@chonknick anything else?
